scripts/ecure.lic

```diff
@@ -115,6 +115,32 @@ else
   @signToUse = nil
 end
 
+def groupcheck
+  group_count = 1
+  group_members = []
+  group_leader = nil
+  group_members.push(Char.name)
+  fput 'GROUP'
+  line = get
+  until line =~ /for a list of other options./
+    if line =~ /You are (?:leading|grouped with) (.*)\./
+      last_match = Regexp.last_match(1)
+      last_match.split(/,(?:\s+and)?| and /).map { |x| x.gsub(/ (?:who is \w+|\(\w+\))/, '').strip }.each { |member|
+        group_members.push(member)
+        group_count += 1
+      }
+      if (leader = last_match.split(/,(?:\s+and)?| and /).find { |x| x =~ /who is leading|\(leading\)/ })
+        group_leader = leader.gsub(/ (?:who is \w+|\(\w+\))/, '').strip
+      else
+        group_leader = Char.name
+      end
+    end
+    group_leader = Char.name if line =~ /^You are not currently in a group\.$/
+    line = get
+  end
+  return [group_members, group_leader, group_count]
+end
+
```

Comment:
Not necessary, can correct the underlying original function much easier than adding a new method.